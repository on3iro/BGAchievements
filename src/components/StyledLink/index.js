/**
 * A simple styled react-router Link element.
  */
import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

export const LinkWrapper = props => {
  // Pulls off secondary prop to avoid react-unknown-prop warning on underlying <a/>
  const { secondary, ...rest } = props

  return <Link {...rest} />
}

const StyledLink = styled(LinkWrapper)`
  display: inline-block;
  text-decoration: none;
  color: ${props => props.secondary ? props.theme.main : props.theme.noneWhite};
  cursor: pointer;
  outline: 0;

  &:visited {
    color: ${props => props.secondary ? props.theme.lightText : props.theme.noneWhite};
  }

  &:hover {
    color: ${props => props.secondary ? props.theme.simple : props.theme.noneWhite};
  }
`

export default StyledLink
