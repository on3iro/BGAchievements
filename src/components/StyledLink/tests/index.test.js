import React from 'react'
import { shallow } from 'enzyme'
import { Link } from 'react-router-dom'

import StyledLink, { LinkWrapper } from '../index'

describe('<StyledLink />', () => {
  it('should render <LinkWrapper />', () => {
    const wrapper = shallow(<StyledLink to='/' />)
    expect(wrapper.find(LinkWrapper).length).toMatchSnapshot()
  })

  describe('<LinkWrapper />', () => {
    it('should render <Link />', () => {
      const wrapper = shallow(<LinkWrapper to='/' />)

      expect(wrapper.find(Link).exists()).toBe(true)
    })
  })
})
