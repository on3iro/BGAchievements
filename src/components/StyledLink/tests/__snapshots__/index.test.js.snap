exports[`<StyledLink /> should render <LinkWrapper /> 1`] = `1`
